Amplify/Core/Support/AmplifyOperation.swift

```diff
@@ -154,6 +160,10 @@ open class AmplifyOperation<Request: AmplifyOperationRequest, Success, Failure:
         )
         publish(result: .failure(cancellationError))
 #endif
+        if let token = resultListenerUnsubscribeToken {
+            unsubscribe(token: token)
+            resultListenerUnsubscribeToken = nil
```

Comment:
redundant `resultListenerUnsubscribeToken = nil` here, resultListenerUnsubscribeToken is being set to `nil` in `subscribe(token: token)` call

Reply:
This is the `unsubscribe` function so this will is intentional. It is called when an Operation is cancelled which could be due to the caused by the `ChildTask` being cancelled. The `subscribe` function which only will remove the listener after the result is received, but it does not set the stored property for `resultListenerUnsubscribeToken` to nil. The cancel function can be called before a result is received and should prevent the listener from being sent a value.  The nil state of `resultListenerUnsubscribeToken` is also used to indicate the listener was removed from the Hub for the `unsubscribe` and `cancel` functions.

Reply:
sorry i meant `unsubscribe(token)` calls `resultListenerUnsubscribeToken = nil` as well as this `cancel()` method calls `unsubscribe(token)` and `resultListenerUnsubscribeToken = nil`